python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -802,18 +909,23 @@ def process_graph(args):
         for name, edata in edge_data[etype].items():
             g.edges[etype].data[name] = th.tensor(edata)
 
-    if args.output_format == "DistDGL":
+    if args.output_format == "DistDGL" and args.num_partitions == 1:
         dgl.distributed.partition_graph(g, args.graph_name, args.num_partitions,
                                         args.output_dir, part_method="None")
+    elif args.output_format == "DistDGL":
+        dgl.distributed.partition_graph(g, args.graph_name, args.num_partitions,
+                                        args.output_dir, part_method="metis")
     elif args.output_format == "DGL":
         dgl.save_graphs(os.path.join(args.output_dir, args.graph_name + ".dgl"), [g])
     else:
         raise ValueError('Unknown output format: {}'.format(args.output_format))
     for ntype in node_id_map:
-        map_data = {}
-        map_data["orig"] = np.array(list(node_id_map[ntype].keys()))
-        map_data["new"] = np.array(list(node_id_map[ntype].values()))
-        write_data_parquet(map_data, os.path.join(args.output_dir, ntype + "_id_remap.parquet"))
+        kv_pairs = node_id_map[ntype].get_key_vals()
+        if kv_pairs is not None:
+            map_data = {}
+            map_data["orig"], map_data["new"] = kv_pairs
+            write_data_parquet(map_data, os.path.join(args.output_dir,
```

Comment:
Can user choose the data format?
Or can we use the same format as input data?

Reply:
not sure what is the best option here. maybe we should allow users to provide an argument to specify the format. or maybe JSON is the best format here.

Reply:
I agree.